### PR TITLE
Container tag rename

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Rename the SUSE Manager containers to match `suse/manager/<version>/<arch>/<server|proxy>-`

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- The documentation will need to use these new paths for proxy and server container images.

- [X] **DONE**

## Test coverage
- No tests: will be covered by end-to-end tests

- [X] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23530

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by mcalmer** [comment]
Labels like this should work with SCC, but it may have downsides.
If server/proxy would be a directory, scc could define the acl like

   `suse/manager/<version>/<arch>/<server|proxy>/**`

And it would match everything in that dir and all subdirs.
If server and proxy are the beginning of the name you can only match in the directory and all images must start with it.

Maybe check back with SCC Team

**Comment by deneb-alpha** [comment]
> Requires a patch similar to this one to be applied on the kiwi metainfo helper:
> 
> ```diff
> Index: kiwi_metainfo_helper
> ===================================================================
> --- kiwi_metainfo_helper	(revision c3b2086c7108efecffd4493ce01fc7e0)
> +++ kiwi_metainfo_helper	(working copy)
> @@ -11,6 +11,7 @@
>  BUILD_DATA="${BUILD_DIST/.dist/.data}"
>  if [ -e "${BUILD_DATA}" ]; then
>  	. "${BUILD_DATA}"
> +	BUILD_ARCH="${BUILD_ARCH%%:*}"
>  
>  	# The build script renames the recipe (to strip _service:foo:), but doesn't update .data
>  	RECIPEFILE="${RECIPEFILE##*:}"
> @@ -25,6 +26,7 @@
>  	echo "Warning: No build data found - chroot build?"
>  	DISTURL="local"
>  	RELEASE=0
> +	BUILD_ARCH="noarch"
>  
>  	# Guess the build recipe
>  	files=(*.kiwi Dockerfile* Chart.yaml*)
> @@ -129,4 +131,5 @@
>      -e "s#%DISTURL%#${DISTURL}#g" \
>      -e "s#%SOURCEURL%#${SOURCEURL}#g" \
>      -e "s/%RELEASE%/${RELEASE}/g" \
> +    -e "s/%ARCH%/${BUILD_ARCH}/g" \
>      -e "s/%BUILDTIME%/$(date --utc +%FT%T.%NZ)/g" "${files[@]}"
> Index: obs-service-kiwi_metainfo_helper.spec
> ===================================================================
> --- obs-service-kiwi_metainfo_helper.spec	(revision c3b2086c7108efecffd4493ce01fc7e0)
> +++ obs-service-kiwi_metainfo_helper.spec	(working copy)
> @@ -1,7 +1,7 @@
>  #
>  # spec file for package obs-service-kiwi_metainfo_helper
>  #
> -# Copyright (c) 2023 SUSE LLC
> +# Copyright (c) 2024 SUSE LLC
>  #
>  # All modifications and additions to the file contributed by third parties
>  # remain the property of their copyright owners, unless otherwise agreed
> Index: test.sh
> ===================================================================
> --- test.sh	(revision c3b2086c7108efecffd4493ce01fc7e0)
> +++ test.sh	(working copy)
> @@ -25,6 +25,7 @@
>  DISTURL="obs://build.opensuse.org/openSUSE:Factory/images/0f40c57dd619e1dff9e512949b6bca09-opensuse-tumbleweed-image:docker"
>  RELEASE=4.2
>  RECIPEFILE=_service:foobar:Dockerfile
> +BUILD_ARCH=aarch64:aarch64_ilp32:armv8l
>  EOF
>  export BUILD_DIST=.dist
>  
> @@ -33,6 +34,7 @@
>  SOURCEURL=%SOURCEURL%
>  RELEASE=%RELEASE%
>  BUILDTIME=%BUILDTIME%
> +ARCH=%ARCH%
>  OS_VERSION=%OS_VERSION%
>  OS_VERSION_NO_DASH=%OS_VERSION_NO_DASH%
>  OS_VERSION_ID=%OS_VERSION_ID%
> @@ -51,6 +53,7 @@
>  SOURCEURL=https://build.opensuse.org/package/show/openSUSE:Factory/opensuse-tumbleweed-image?rev=0f40c57dd619e1dff9e512949b6bca09
>  RELEASE=4.2
>  BUILDTIME=2018-10-30T09:19:02.074934628Z
> +ARCH=aarch64
>  OS_VERSION=15-SP4
>  OS_VERSION_NO_DASH=15 SP4
>  OS_VERSION_ID=15.4
> @@ -67,6 +70,7 @@
>  DISTURL="obs://build.suse.de/SUSE:SLE-15-SP3:Update:CR/images/5f0a221b7877396cbf977205e64690d2-sles15-image"
>  RELEASE=4.2
>  RECIPEFILE=_service:foobar:Dockerfile
> +BUILD_ARCH=aarch64:aarch64_ilp32:armv8l
>  EOF
>  
>  cat >Dockerfile <<EOF
> @@ -89,6 +93,7 @@
>  SOURCEURL=%SOURCEURL%
>  RELEASE=%RELEASE%
>  BUILDTIME=%BUILDTIME%
> +ARCH=%ARCH%
>  EOF
>  
>  bash "${script}"
> @@ -98,4 +103,5 @@
>  SOURCEURL=https://local/package/show/local/local?rev=local
>  RELEASE=0
>  BUILDTIME=2018-10-30T09:19:02.074934628Z
> +ARCH=noarch
>  EOF
> ```

@cbosdo I saw the discussion with Fabian [here](https://suse.slack.com/archives/C02DR0C5XUY/p1706525728762339).
But I didn't get  completely if your patch can  be submitted or if we need some special custom version only for us

File: containers/server-hub-xmlrpc-api-image/server-hub-xmlrpc-api-image.changes
[the remark this answers is not in the document]
Comment (cbosdo) [reply containers/server-hub-xmlrpc-api-image/server-hub-xmlrpc-api-image.changes:1]: I can do so test pushes on my home branch to verify that

**Comment by cbosdo** [comment]
> @cbosdo I saw the discussion with Fabian [here](https://suse.slack.com/archives/C02DR0C5XUY/p1706525728762339). But I didn't get completely if your patch can be submitted or if we need some special custom version only for us

To be honest I didn't understand either...

**Comment by cbosdo** [comment]
> Labels like this should work with SCC, but it may have downsides. If server/proxy would be a directory, scc could define the acl like
> 
> `suse/manager/<version>/<arch>/<server|proxy>/**`
> 
> And it would match everything in that dir and all subdirs. If server and proxy are the beginning of the name you can only match in the directory and all images must start with it.

I'm not sure we want subdirs there... unless we want a special folder for migration images like: `suse/manager/<version>/<arch>/server/migration/pgsql-14-16`. But we should only have one image (maybe two for Uyuni) for migration for each version.

**Comment by mcalmer** [comment]
> > Labels like this should work with SCC, but it may have downsides. If server/proxy would be a directory, scc could define the acl like
> > `suse/manager/<version>/<arch>/<server|proxy>/**`
> > And it would match everything in that dir and all subdirs. If server and proxy are the beginning of the name you can only match in the directory and all images must start with it.
> 
> I'm not sure we want subdirs there... unless we want a special folder for migration images like: `suse/manager/<version>/<arch>/server/migration/pgsql-14-16`. But we should only have one image (maybe two for Uyuni) for migration for each version.

Confidential Compute will add also containers. And in future versions we want to split out more.
But if we are fine that all images must start with `server-` or `proxy-` and we do not plan to use subdirectories, we can go this way.

**Comment by cbosdo** [comment]
> > > Labels like this should work with SCC, but it may have downsides. If server/proxy would be a directory, scc could define the acl like
> > > `suse/manager/<version>/<arch>/<server|proxy>/**`
> > > And it would match everything in that dir and all subdirs. If server and proxy are the beginning of the name you can only match in the directory and all images must start with it.
> > 
> > 
> > I'm not sure we want subdirs there... unless we want a special folder for migration images like: `suse/manager/<version>/<arch>/server/migration/pgsql-14-16`. But we should only have one image (maybe two for Uyuni) for migration for each version.
> 
> Confidential Compute will add also containers. And in future versions we want to split out more. But if we are fine that all images must start with `server-` or `proxy-` and we do not plan to use subdirectories, we can go this way.

I don't care that much... It's just a `-` replaced by a `/`... but it involves more mess for release engineers and tools and `main.tf`s tweaking to change all images rather than one that is still not used.

**Comment by rjmateus** [comment]
From my point of view, we should use a sub-directory. This way is clear that all images should be under it and we are not playing with a name prefix that can influence image accessibility. It would make it less error-prone if someone in the future adds a new image and forgets the image name prefix is controlling the user access.

**Comment by deneb-alpha** [comment]
Changes on OBS/IBS:
* devel project: https://build.opensuse.org/request/show/1144539
* Factory: https://build.opensuse.org/request/show/1144543
* SLE: https://build.suse.de/request/show/320686

File: containers/server-hub-xmlrpc-api-image/server-hub-xmlrpc-api-image.changes
[the remark this answers is not in the document]
Comment (cbosdo) [reply containers/server-hub-xmlrpc-api-image/server-hub-xmlrpc-api-image.changes:1]: I ran `git tag -s -m "Tag alias for 5.0.2 server-hub-xmlrpc-api-image" server-hub-xmlrpc-api-image-5.0.2 85f000479511172b03a20107b0a095f638b63355` before running `tito tag` in the `server-hub-xml-rpc-api-image` folder and it worked.

**Pull request merged**